### PR TITLE
WIP (windows support): stub for supporting windows via fsnotify.v1

### PR DESCRIPTION
**Pull request by blaggacao**
No idea, if this works, nor how to build it, so far. But it is a starting point to the best of my knowledge. I would be pleased about some sort of collaboration in order to get this badly needed docker-rsync working on windows...

### CODE REVIEW

**Comment by blaggacao** [comment]
Feels good, it built successfully! :dancer:

**Comment by blaggacao** [comment]
To last commit: I didn't find a more elegant way to avoid this. Maybe this "external dependency" can be removed completely, although a basic one?

**Comment by mattes** [comment]
Thanks, @blaggacao for your PR. I don't want to duplicate main.go. I think we should look at watch.go. I will have a look into this later this week.

**Comment by blaggacao** [comment]
Actually, I found watch.go to be the part that is taken care of by fsnotify. An alternative is to make a wrapper around fsnotify in order to adhere to the watch.go signature. However, this would be a wrapper around a wrapper. Until fsevents gets merged into fsnotify, this would be temporary anyhow...

However, I think fsnotify project misses some traction atm.

File: main_windows.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+			fmt.Printf("error: unable to get port for machine '%v': %v\n", machineName, err)
+			os.Exit(1)
+		}
+
+		Provision(machineName, *verbose)
+		RunSSHCommand(machineName, "sudo mkdir -p "+rpathDir, *verbose)
+		fmt.Printf("Syncing %s (local) to %s (docker-machine %s)\n", *srcpath, *dstpath, machineName)
+		Sync(machineName, port, rpath, rpathDir, *verbose) // initial sync
+
+		if *watch {
```
Comment (blaggacao) [inline main_windows.go]: This is the only differing part. I also was not pleased to duplicate whole main.go - Maybe there could be a small wrapper around the actual watch go, which imitates fsnotify.v1 signature?

File: rsync.go [hunk 2]
```diff
@@ -43,8 +43,8 @@ func Sync(via string, port uint, src, dst string, verbose bool) {
 
 	command := "rsync " + strings.Join(args, " ")
 
-	// fmt.Println("/bin/sh", "-c", command)
-	cmd := exec.Command("/bin/sh", "-c", command)
+	// fmt.Println("sh", "-c", command)
```
Comment (blaggacao) [inline rsync.go]: Can we remove this dependency on sh? Or replace it by a corresponding multiplattfrom go library?

**Comment by duro** [comment]
Hey @blaggacao and @mattes, any progress on this? I have been using this library with great success on my OS X machine, but I am about to start a new project with a team where some of the engineers use Windows.

Seems like you guys are close, and I'd love to be able to make this library part of the tooling for this new project as well.

**Comment by blaggacao** [comment]
@duro I'm afraid, I'm out, as this is no longer a priority for me.

**Comment by duro** [comment]
@blaggacao did you go a different route?

**Comment by blaggacao** [comment]
Well I'm betting on docker-machine to leverage the docker volume plugin to setup a net protocol file share or an rsync volume.

So syncing would be handled by docker.
And mounting would be handled by a `docker-machine share`.

I've seen a proposal, but has no very high prio, I think.

**Comment by enmand** [comment]
I've added support for Windows in https://github.com/synack/docker-rsync/pull/13.

-Dan

**Comment by mattes** [comment]
Closing in favor of #13

**Pull request closed without merging**